swagger/src/main/scala/org/http4s/rho/swagger/SwaggerModelsBuilder.scala

```diff
@@ -240,20 +250,27 @@ private[swagger] class SwaggerModelsBuilder(formats: SwaggerFormats) {
       }
     }
 
-    def mkCollectionProperty(tpe: Type): Property = {
+    def mkCollectionProperty(tpe: Type): Option[Property] = {
       val param = tpe.dealias.typeArgs.head
       val prop =
         if (param.isPrimitive)
-          mkPrimitiveProperty(param)
+          mkPrimitiveProperty(param).some
         else if (param.isCollection)
           typeToProp(param)
         else
-          RefProperty(ref = param.simpleName)        
+          RefProperty(ref = param.simpleName).some 
 
-      ArrayProperty(items = prop)
+      prop.map(p => ArrayProperty(items = p))
     }
 
-    code -> Response(description = descr, schema = otpe.map(typeToProp))
+    val schema = {
+      try otpe.flatMap(typeToProp)
+      catch { case _: Throwable =>
```

Comment:
Yes, in hindsight maybe NonFatal would have been better. Perhaps logging
the error or at least the message and type would be good too.
On Apr 1, 2015 4:04 PM, "Leif Warner" <notifications@github.com> wrote:

> In
> swagger/src/main/scala/org/http4s/rho/swagger/SwaggerModelsBuilder.scala
> <https://github.com/http4s/rho/pull/71#discussion_r27606036>:
>
> >      }
> >
> > -    code -> Response(description = descr, schema = otpe.map(typeToProp))
> > +    val schema = {
> > +      try otpe.flatMap(typeToProp)
> > +      catch { case _: Throwable =>
>
> Wouldn't we want a more specific catch here? At least not Throwable?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/http4s/rho/pull/71/files#r27606036>.
>